<details>
<summary>libvirt doesn't build (with sandbox enabled) on x86_64-darwin. It doesn't build on master</summary>

```
error: builder for '/nix/store/7h0ma9nwqn2kvk2lazq6w2mrwslf65ip-python3.11-nixops-gce-unstable-2023-05-26.drv' failed with exit code 1;
       last 10 log lines:
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_gcp-1.0-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_gcp-1.0-py3-none-any.whl
       >   - cryptography==40.0.1 not satisfied by version 42.0.5
       For full logs, run 'nix log /nix/store/7h0ma9nwqn2kvk2lazq6w2mrwslf65ip-python3.11-nixops-gce-unstable-2023-05-26.drv'.
error: builder for '/nix/store/d3z4nx44bqzj2l2d5zga6q1p1mfbg4rc-python3.11-nixops-hetzner-unstable-2022-04-23.drv' failed with exit code 1;
       last 10 log lines:
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_hetzner-1.0-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_hetzner-1.0-py3-none-any.whl
       >   - typing-extensions<4.0.0,>=3.7.4 not satisfied by version 4.11.0
       For full logs, run 'nix log /nix/store/d3z4nx44bqzj2l2d5zga6q1p1mfbg4rc-python3.11-nixops-hetzner-unstable-2022-04-23.drv'.
error: builder for '/nix/store/4pay3bg9f3bjgdgbdmb8j6pjkfyffzy1-python3.11-nixops-hetznercloud-unstable-2023-02-19.drv' failed with exit code 1;
       last 10 log lines:
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_hetznercloud-0.1.3-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_hetznercloud-0.1.3-py3-none-any.whl
       >   - hcloud==1.18.2 not satisfied by version 1.35.0
       >   - typing-extensions<4.0.0,>=3.7.4 not satisfied by version 4.11.0
       For full logs, run 'nix log /nix/store/4pay3bg9f3bjgdgbdmb8j6pjkfyffzy1-python3.11-nixops-hetznercloud-unstable-2023-02-19.drv'.
error: builder for '/nix/store/jcdw0nxwwn530l4gbzipin0dwz5qhpav-python3.11-qemu-0.6.1.0a1.drv' failed with exit code 1;
       last 10 log lines:
       > Traceback (most recent call last):
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/bin/.qom-wrapped", line 6, in <module>
       >     from qemu.utils.qom import main
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/lib/python3.11/site-packages/qemu/utils/qom.py", line 41, in <module>
       >     from .qom_fuse import QOMFuse
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/lib/python3.11/site-packages/qemu/utils/qom_fuse.py", line 48, in <module>
       >     import fuse
       >   File "/nix/store/2gkw9jgh3f533lydjsf3n1ipxwdzhglx-python3.11-fusepy-3.0.1/lib/python3.11/site-packages/fuse.py", line 115, in <module>
       >     raise EnvironmentError('Unable to find libfuse')
       > OSError: Unable to find libfuse
       For full logs, run 'nix log /nix/store/jcdw0nxwwn530l4gbzipin0dwz5qhpav-python3.11-qemu-0.6.1.0a1.drv'.
error: builder for '/nix/store/7s782bsqjapyzf0fn303x68wm5cwzjbd-diffoscope-265.drv' failed with exit code 1;
       last 10 log lines:
       >     def test_diff(differences):
       >         expected_diff = get_data("wasm_expected_diff")
       > >       actual_diff = differences[0].unified_diff
       > E       IndexError: list index out of range
       >
       > tests/comparators/test_wasm.py:50: IndexError
       > =========================== short test summary info ============================
       > FAILED tests/comparators/test_wasm.py::test_diff - IndexError: list index out of range
       > = 1 failed, 624 passed, 55 skipped, 15 deselected, 1 xfailed, 1 xpassed in 7929.02s (2:12:09) =
       > /nix/store/79qly7vi6mjpmbmgivn4dfzl69f0f1sy-stdenv-darwin/setup: line 1588: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/7s782bsqjapyzf0fn303x68wm5cwzjbd-diffoscope-265.drv'.
error: builder for '/nix/store/08vypajjpzwjvnaavyixc55m6dqhb4kx-libvirt-10.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >
       >
       > Ok:                 108
       > Expected Fail:      0
       > Fail:               0
       > Unexpected Pass:    0
       > Skipped:            13
       > Timeout:            1
       >
       > Full log written to /private/tmp/nix-build-libvirt-10.0.0.drv-0/source/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/08vypajjpzwjvnaavyixc55m6dqhb4kx-libvirt-10.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/z32kp9vrfih3dknv42yr1fhhbw15wc8i-libvirt-glib-5.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/53bx2n82bcaz5syn64vs5l4h8xvm8dqd-mgmt-unstable-2022-10-24.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4y9sj44dfa9mi587mc5pvab6r9q76bgy-ocaml-libvirt-0.6.1.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mhalk47sj12mvdzs3izdwzwbmjjq9ik3-python3.11-libvirt-10.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pxkp9g9lzgb262j7nx67ssm5mhylkha2-ruby3.1-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cns1i5nmwcic7gyrnhc20ah3prcfknp1-ruby3.2-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sy0a9yilhpd5ajrq4rl4zjqxzkbjaslw-ruby3.3-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vl5cbc0ihds0cl36n081vynvr6rjf68s-python3.11-nixops-libvirtd-unstable-2023-09-01.drv' failed to build
error: 3 dependencies of derivation '/nix/store/ma6crmv3r1rp9m7016w8ncw0j7zqkkb1-virt-manager-4.1.0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/92i3g6znggb6972c27h8igic7scaz4d4-virt-viewer-11.0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/c1agaklfgpbc4bf9bk6y1bdizqsgqfmx-virt-viewer-11.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sk9skjphqjppy1k01p2gnf1mgcn14zxb-appvm-unstable-2021-12-20.drv' failed to build
error: 4 dependencies of derivation '/nix/store/mkjn9ny88zjn0v4y3i7g8s1x78hqinqs-nixops-unstable-2024-02-28.drv' failed to build
error: builder for '/nix/store/535c3f1zvw6gypijm3by9q3i9xg2bsgf-python3.12-cot-2.2.1.drv' failed with exit code 1;
       last 10 log lines:
       >     return get_versions()["version"]
       >            ^^^^^^^^^^^^^^
       >   File "/private/tmp/nix-build-python3.12-cot-2.2.1.drv-0/cot-2.2.1/versioneer.py", line 1408, in get_versions
       >     cfg = get_config_from_root(root)
       >           ^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/private/tmp/nix-build-python3.12-cot-2.2.1.drv-0/cot-2.2.1/versioneer.py", line 342, in get_config_from_root
       >     parser = configparser.SafeConfigParser()
       >              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       > AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
       > /nix/store/79qly7vi6mjpmbmgivn4dfzl69f0f1sy-stdenv-darwin/setup: line 1588: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/535c3f1zvw6gypijm3by9q3i9xg2bsgf-python3.12-cot-2.2.1.drv'.
error: 16 dependencies of derivation '/nix/store/h6grs7gbcvflpknipid22d1pyafrbljs-review-shell.drv' failed to build
```

```
nix build nixpkgs#libvirt --out-link /dev/null --rebuild
error: builder for '/nix/store/fwngvfmylb17js2p7r96fkj5s2phfj87-libvirt-10.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >
       >
       > Ok:                 108
       > Expected Fail:      0
       > Fail:               0
       > Unexpected Pass:    0
       > Skipped:            13
       > Timeout:            1
       >
       > Full log written to /private/tmp/nix-build-libvirt-10.0.0.drv-1/source/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/fwngvfmylb17js2p7r96fkj5s2phfj87-libvirt-10.0.0.drv'.
```
</details>

Other error are not related. I think diffoscope is already broken in master (or wait for hydra)

--- a/pkgs/applications/virtualization/qemu/default.nix
+++ b/pkgs/applications/virtualization/qemu/default.nix
@@ -55,11 +55,11 @@ stdenv.mkDerivation (finalAttrs: {
     + lib.optionalString hostCpuOnly "-host-cpu-only"
     + lib.optionalString nixosTestRunner "-for-vm-tests"
     + lib.optionalString toolsOnly "-utils";
-  version = "8.2.4";
+  version = "9.0.0";
 
   src = fetchurl {
     url = "https://download.qemu.org/qemu-${finalAttrs.version}.tar.xz";
-    hash = "sha256-7PVTf+q5JkG5nXSC9VHyGV06W9NKzvnVK/v/NTpgc5c=";
+    hash = "sha256-MnCKxmww2MiSYz6paMdxwcdtWX1w3erSGg0izPOG2mk=";
   };
 
   depsBuildBuild = [ buildPackages.stdenv.cc ]
@@ -124,11 +124,9 @@ stdenv.mkDerivation (finalAttrs: {
     # Cocoa clipboard support only works on macOS 10.14+
     ./revert-ui-cocoa-add-clipboard-support.patch
     # Standard about panel requires AppKit and macOS 10.13+
-    (fetchpatch {
-      url = "https://gitlab.com/qemu-project/qemu/-/commit/99eb313ddbbcf73c1adcdadceba1423b691c6d05.diff";
-      sha256 = "sha256-gTRf9XENAfbFB3asYCXnw4OV4Af6VE1W56K2xpYDhgM=";
-      revert = true;
-    })
+    ./revert-ui-cocoa-use-the-standard-about-panel.patch
+    # Safe area insets require macOS 11+
+    ./remove-ui-cocoa-use-safe-area-insets.patch
     # Workaround for upstream issue with nested virtualisation: https://gitlab.com/qemu-project/qemu/-/issues/1008
     (fetchpatch {
       url = "https://gitlab.com/qemu-project/qemu/-/commit/3e4546d5bd38a1e98d4bd2de48631abf0398a3a2.diff";
@@ -239,6 +237,10 @@ stdenv.mkDerivation (finalAttrs: {
     # get-fsinfo attempts to access block devices, disallowed by sandbox
     sed -i -e '/\/qga\/get-fsinfo/d' -e '/\/qga\/blacklist/d' \
       ../tests/unit/test-qga.c
+
+    # xattrs are not allowed in the sandbox
+    substituteInPlace ../tests/qtest/virtio-9p-test.c \
+      --replace-fail mapped-xattr mapped-file
   '' + lib.optionalString stdenv.isDarwin ''
     # skip test that stalls on darwin, perhaps due to subtle differences
     # in fifo behaviour

--- a/pkgs/applications/virtualization/qemu/remove-ui-cocoa-use-safe-area-insets.patch
+++ b/pkgs/applications/virtualization/qemu/remove-ui-cocoa-use-safe-area-insets.patch
@@ -1,0 +1,14 @@
+diff --git a/ui/cocoa.m b/ui/cocoa.m
+index 25e0db9dd0..7ce889d798 100644
+--- a/ui/cocoa.m
++++ b/ui/cocoa.m
+@@ -539,9 +539,6 @@ - (NSSize)fixAspectRatio:(NSSize)max
+ - (NSSize) screenSafeAreaSize
+ {
+     NSSize size = [[[self window] screen] frame].size;
+-    NSEdgeInsets insets = [[[self window] screen] safeAreaInsets];
+-    size.width -= insets.left + insets.right;
+-    size.height -= insets.top + insets.bottom;
+     return size;
+ }
+ 

--- a/pkgs/applications/virtualization/qemu/revert-ui-cocoa-add-clipboard-support.patch
+++ b/pkgs/applications/virtualization/qemu/revert-ui-cocoa-add-clipboard-support.patch
@@ -47,7 +47,7 @@ index 660d3e0935..0e6760c360 100644
 -static QemuClipboardInfo *cbinfo;
 -static QemuEvent cbevent;
 -
- // Utility functions to run specified code block with iothread lock held
+ // Utility functions to run specified code block with the BQL held
  typedef void (^CodeBlock)(void);
  typedef bool (^BoolCodeBlock)(void);
 @@ -1799,107 +1794,6 @@ static void addRemovableDevicesMenuItems(void)
@@ -65,7 +65,7 @@ index 660d3e0935..0e6760c360 100644
 -        return;
 -    }
 -
--    with_iothread_lock(^{
+-    with_bql(^{
 -        QemuClipboardInfo *info = qemu_clipboard_info_ref(cbinfo);
 -        qemu_event_reset(&cbevent);
 -        qemu_clipboard_request(info, QEMU_CLIPBOARD_TYPE_TEXT);
@@ -73,9 +73,9 @@ index 660d3e0935..0e6760c360 100644
 -        while (info == cbinfo &&
 -               info->types[QEMU_CLIPBOARD_TYPE_TEXT].available &&
 -               info->types[QEMU_CLIPBOARD_TYPE_TEXT].data == NULL) {
--            qemu_mutex_unlock_iothread();
+-            bql_unlock();
 -            qemu_event_wait(&cbevent);
--            qemu_mutex_lock_iothread();
+-            bql_lock();
 -        }
 -
 -        if (info == cbinfo) {
@@ -186,9 +186,9 @@ index 660d3e0935..0e6760c360 100644
  }
  
 @@ -2071,12 +1952,6 @@ static void cocoa_display_init(DisplayState *ds, DisplayOptions *opts)
- 
      // register vga output callbacks
      register_displaychangelistener(&dcl);
+     [cocoaView updateUIInfo];
 -
 -    qemu_event_init(&cbevent, false);
 -    cbowner = [[QemuCocoaPasteboardTypeOwner alloc] init];

--- a/pkgs/applications/virtualization/qemu/revert-ui-cocoa-use-the-standard-about-panel.patch
+++ b/pkgs/applications/virtualization/qemu/revert-ui-cocoa-use-the-standard-about-panel.patch
@@ -1,0 +1,145 @@
+diff --git a/ui/cocoa.m b/ui/cocoa.m
+index 25e0db9dd0..4af0712036 100644
+--- a/ui/cocoa.m
++++ b/ui/cocoa.m
+@@ -93,6 +93,7 @@ static void cocoa_switch(DisplayChangeListener *dcl,
+ 
+ static void cocoa_refresh(DisplayChangeListener *dcl);
+ 
++static NSWindow *about_window;
+ static const DisplayChangeListenerOps dcl_ops = {
+     .dpy_name          = "cocoa",
+     .dpy_gfx_update = cocoa_update,
+@@ -1180,6 +1181,7 @@ - (void)changeDeviceMedia:(id)sender;
+ - (BOOL)verifyQuit;
+ - (void)openDocumentation:(NSString *)filename;
+ - (IBAction) do_about_menu_item: (id) sender;
++- (void)make_about_window;
+ - (void)adjustSpeed:(id)sender;
+ @end
+ 
+@@ -1227,6 +1229,8 @@ - (id) init
+         [pauseLabel setFont: [NSFont fontWithName: @"Helvetica" size: 90]];
+         [pauseLabel setTextColor: [NSColor blackColor]];
+         [pauseLabel sizeToFit];
++
++        [self make_about_window];
+     }
+     return self;
+ }
+@@ -1549,29 +1553,92 @@ - (BOOL)verifyQuit
+ /* The action method for the About menu item */
+ - (IBAction) do_about_menu_item: (id) sender
+ {
+-    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+-    char *icon_path_c = get_relocated_path(CONFIG_QEMU_ICONDIR "/hicolor/512x512/apps/qemu.png");
+-    NSString *icon_path = [NSString stringWithUTF8String:icon_path_c];
+-    g_free(icon_path_c);
+-    NSImage *icon = [[NSImage alloc] initWithContentsOfFile:icon_path];
+-    NSString *version = @"QEMU emulator version " QEMU_FULL_VERSION;
+-    NSString *copyright = @QEMU_COPYRIGHT;
+-    NSDictionary *options;
+-    if (icon) {
+-        options = @{
+-            NSAboutPanelOptionApplicationIcon : icon,
+-            NSAboutPanelOptionApplicationVersion : version,
+-            @"Copyright" : copyright,
+-        };
+-        [icon release];
+-    } else {
+-        options = @{
+-            NSAboutPanelOptionApplicationVersion : version,
+-            @"Copyright" : copyright,
+-        };
+-    }
+-    [NSApp orderFrontStandardAboutPanelWithOptions:options];
+-    [pool release];
++    [about_window makeKeyAndOrderFront: nil];
++}
++
++/* Create and display the about dialog */
++- (void)make_about_window
++{
++    /* Make the window */
++    int x = 0, y = 0, about_width = 400, about_height = 200;
++    NSRect window_rect = NSMakeRect(x, y, about_width, about_height);
++    about_window = [[NSWindow alloc] initWithContentRect:window_rect
++                    styleMask:NSWindowStyleMaskTitled | NSWindowStyleMaskClosable |
++                    NSWindowStyleMaskMiniaturizable
++                    backing:NSBackingStoreBuffered
++                    defer:NO];
++    [about_window setTitle: @"About"];
++    [about_window setReleasedWhenClosed: NO];
++    [about_window center];
++    NSView *superView = [about_window contentView];
++
++    /* Create the dimensions of the picture */
++    int picture_width = 80, picture_height = 80;
++    x = (about_width - picture_width)/2;
++    y = about_height - picture_height - 10;
++    NSRect picture_rect = NSMakeRect(x, y, picture_width, picture_height);
++
++    /* Make the picture of QEMU */
++    NSImageView *picture_view = [[NSImageView alloc] initWithFrame:
++                                                     picture_rect];
++    char *qemu_image_path_c = get_relocated_path(CONFIG_QEMU_ICONDIR "/hicolor/512x512/apps/qemu.png");
++    NSString *qemu_image_path = [NSString stringWithUTF8String:qemu_image_path_c];
++    g_free(qemu_image_path_c);
++    NSImage *qemu_image = [[NSImage alloc] initWithContentsOfFile:qemu_image_path];
++    [picture_view setImage: qemu_image];
++    [picture_view setImageScaling: NSImageScaleProportionallyUpOrDown];
++    [superView addSubview: picture_view];
++
++    /* Make the name label */
++    NSBundle *bundle = [NSBundle mainBundle];
++    if (bundle) {
++        x = 0;
++        y = y - 25;
++        int name_width = about_width, name_height = 20;
++        NSRect name_rect = NSMakeRect(x, y, name_width, name_height);
++        NSTextField *name_label = [[NSTextField alloc] initWithFrame: name_rect];
++        [name_label setEditable: NO];
++        [name_label setBezeled: NO];
++        [name_label setDrawsBackground: NO];
++        [name_label setAlignment: NSTextAlignmentCenter];
++        NSString *qemu_name = [[bundle executablePath] lastPathComponent];
++        [name_label setStringValue: qemu_name];
++        [superView addSubview: name_label];
++    }
++
++    /* Set the version label's attributes */
++    x = 0;
++    y = 50;
++    int version_width = about_width, version_height = 20;
++    NSRect version_rect = NSMakeRect(x, y, version_width, version_height);
++    NSTextField *version_label = [[NSTextField alloc] initWithFrame:
++                                                      version_rect];
++    [version_label setEditable: NO];
++    [version_label setBezeled: NO];
++    [version_label setAlignment: NSTextAlignmentCenter];
++    [version_label setDrawsBackground: NO];
++
++    /* Create the version string*/
++    NSString *version_string;
++    version_string = [[NSString alloc] initWithFormat:
++    @"QEMU emulator version %s", QEMU_FULL_VERSION];
++    [version_label setStringValue: version_string];
++    [superView addSubview: version_label];
++
++    /* Make copyright label */
++    x = 0;
++    y = 35;
++    int copyright_width = about_width, copyright_height = 20;
++    NSRect copyright_rect = NSMakeRect(x, y, copyright_width, copyright_height);
++    NSTextField *copyright_label = [[NSTextField alloc] initWithFrame:
++                                                        copyright_rect];
++    [copyright_label setEditable: NO];
++    [copyright_label setBezeled: NO];
++    [copyright_label setDrawsBackground: NO];
++    [copyright_label setAlignment: NSTextAlignmentCenter];
++    [copyright_label setStringValue: [NSString stringWithFormat: @"%s",
++                                     QEMU_COPYRIGHT]];
++    [superView addSubview: copyright_label];
+ }
+ 
+ /* Used by the Speed menu items */

